src/override/rust.cc

```diff
@@ -41,14 +41,20 @@ namespace
       if constexpr (size_change == SizeChange::Unknown)
         std::memcpy(p, ptr, old_size < new_size ? old_size : new_size);
       else if constexpr (size_change == SizeChange::Grow)
+      {
+        SNMALLOC_ASSERT(old_size <= new_size);
         std::memcpy(p, ptr, old_size);
+      }
       else
+      {
+        SNMALLOC_ASSERT(new_size <= old_size);
         std::memcpy(p, ptr, new_size);
+      }
       alloc.dealloc(ptr, aligned_old_size);
     }
     return p;
   }
-}
+} // namespace
```

Comment:
This comment looks suspect

Reply:
oh, that actually comes from `clang-tidy`, who was ready to add namespace indicator at the ending of the scope.